tests/integration/tier0/test-check-rollback-handling/test_check_rollback_handling.py

```diff
@@ -104,3 +101,111 @@ def test_check_untrack_pkgs_force(convert2rhel, shell):
         assert c2r.exitstatus != 0
 
     post_rollback_check(shell)
+
+
+def terminate_and_assert_good_rollback(c2r):
+    if "oracle-7" in booted_os or "centos-7" in booted_os:
+        # Use 'Ctrl + c' first to check for unexpected behaviour
+        # of the rollback feature after process termination
+        c2r.sendcontrol("c")
+        # Due to inconsistent behaviour of 'Ctrl + c' on some distros
+        # use Ctrl + d instead
+        c2r.sendcontrol("d")
+        # Assert the rollback finished all tasks by going through its last task
+        assert c2r.expect("Rollback: Removing installed RHSM certificate") == 0
+        assert c2r.exitstatus != 1
+    else:
+        c2r.sendcontrol("c")
+        assert c2r.expect("Rollback: Removing installed RHSM certificate") == 0
+        assert c2r.exitstatus != 1
+
+
+def test_terminate_registration_asap(convert2rhel):
+    # Send termination signal immediately after c2r tries the registration.
+    with convert2rhel(
+        ("-y --no-rpm-va --serverurl {} --username {} --password {}").format(
+            env.str("RHSM_SERVER_URL"),
+            env.str("RHSM_USERNAME"),
+            env.str("RHSM_PASSWORD"),
+        )
+    ) as c2r:
+        c2r.expect("Registering the system using subscription-manager")
+        terminate_and_assert_good_rollback(c2r)
+
+
+def test_terminate_username(convert2rhel):
+    # Send termination signal on the user prompt for Username.
+    with convert2rhel("-y --no-rpm-va") as c2r:
+        c2r.expect("Username:")
+        terminate_and_assert_good_rollback(c2r)
+
+
+def test_terminate_password(convert2rhel):
```

Comment:
`test_terminate_on_password`

Reply:
Sounds better, thank you for suggestion.